File: Data Structures & Algorithms/eating-bananas/submission-3.java
class Solution {
    public int minEatingSpeed(int[] arr, int h) {
        int l = 1;
        int r = Arrays.stream(arr).max().getAsInt();
        int res = r;
        while(l<=r){
            int mid = l+(r-l)/2;
            int count = 0;
            for(int i : arr){
                count+=Math.ceil((double)i/mid);
            }
            if(count<=h){
                res = mid;
                r = mid-1;
            }
            else{
                l = mid+1;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/can-place-flowers/submission-3.py
class Solution:
    def canPlaceFlowers(self, arr: List[int], n: int) -> bool:
        count = 0
        for i in range(len(arr)-1):
            if i==0 and arr[i+1]==0 and arr[i]==0:
                count+=1
                arr[i]=1
            elif arr[i]==0 and arr[i+1]==0 and arr[i-1]==0:
                arr[i]=1
                count+=1
        
        return count>=n

